### ISSUE TITLE
Auto connect to Wi-Fi enterprise network

### ISSUE THREAD

**Issue by jeniha** [Thinstation/thinstation#591]
Hi,
i have two questions:
1.Can set TS to connect automatically to the enterprise wifi-fi network with PEAP protocol and **WPA2-Enterprise**?**_I need to use username and password to connect to our company wireless network._**
2.If i can, what i need to do to make it work?

**Comment by jeniha** [comment]
Is it possible?

**Comment by Thinstation** [comment]
You will have to write a custom wpa_supplicant.conf file. You can then include it in your build and specifiy CUSTOM_SUPPLICANT_CONF=<Full path to file> in .buildtime

**Comment by jeniha** [comment]
Can you give me, a hint what I need to include in conf. I need to use username, ad, and password to log in wireless. Did i need special packages to enable.

**Comment by Thinstation** [comment]
ctrl_interface=/run/wpa_supplicant
ctrl_interface_group=0
eapol_version=2
ap_scan=0
network={
	ssid="example"
	scan_ssid=1
	key_mgmt=WPA-EAP
	eap=PEAP
	identity="user@example.com"
	password="foobar"
	ca_cert="/etc/cert/ca.pem"
	phase1="peaplabel=0"
	phase2="auth=MSCHAPV2"
}

**Comment by Thinstation** [comment]
man wpa_supplicant.conf

**Comment by jeniha** [comment]
Did I need edit in build conf?

**Comment by Thinstation** [comment]
That's an example of a custom supplicant. You would then tell thinstation to use the custom supplicant with
CUSTOM_SUPPLICANT_CONF="path to supplicant" in .buildtime

**Comment by jeniha** [comment]
I use ts now only with ethernet. Did i need to change networkmanager or something else?

**Comment by Thinstation** [comment]
Network manager or autonet. Doesn’t matter.

**Comment by jeniha** [comment]
Hi,
today after 2-3 hours of trying, cant get it to work.For supplicant i use this:
ctrl_interface=/run/wpa_supplicant
ctrl_interface_group=0
eapol_version=2
ap_scan=0
network={
ssid="ssid"
scan_ssid=1
key_mgmt=WPA-EAP
eap=PEAP
identity="username@domain.com"
password="foobar"
ca_cert="/etc/cert/ca.pem"
phase1="peaplabel=0"
phase2="auth=MSCHAPV2"
}
I made one file that i put it in thinstation folder with name wpa_supplicant.conf.In buildtime i put CUSTOM_SUPPLICANT_CONF="/thinstation/wpa_supplicant.conf" on bottom
In build.conf i uncomment package wpa_supplicant and package wireless.Where is my mistake?

BTW:For chiffres  we use AES.

**Comment by Thinstation** [comment]
make a folder in /build/packages/base/ called thinstation, then copy your wpa_supplicant.conf into that folder, then your buildtime and build changes should work.

**Comment by jeniha** [comment]
Nope, not even a try to connecting in event log in AD.

**Comment by Thinstation** [comment]
If you want to use wireless autoconnect with a custom wpa_supplicant, you need to use autonet. If you are using ethernet, you can use networkmanager.

**Comment by jeniha** [comment]
How to enable autonet?

**Comment by Thinstation** [comment]
in build.conf, comment out networkmanager, then add package autonet

**Comment by jeniha** [comment]
Nope. again not even a trying to connect.I think i not even reach wpa_supplicant.With ifconfig i can see wireless adapter...
In build.conf i make this changes:
Enable package autonet
Enable package wpa_supplicant
Enable package wireless

In buildtime i make this changes:
Add in CUSTOM_SUPPLICANT_CONF="/thinstation/wpa_supplicant.conf"

This is in wpa_supplicant :
ctrl_interface=/run/wpa_supplicant
ctrl_interface_group=0
eapol_version=2
ap_scan=0
network={
ssid="ssid"
scan_ssid=1
key_mgmt=WPA-EAP
eap=PEAP
identity="username@domain.com"
password="foobar"
ca_cert="/etc/cert/ca.pem"
phase1="peaplabel=0"
phase2="auth=MSCHAPV2"
}
Where is my mistake?

**Comment by Thinstation** [comment]
CUSTOM_SUPPLICANT_CONF is relative to the running file-system, not the build file-system. make a package for the supplicant, or put it in the etc dir of an existing package and then set CUSTOM_SUPPLICANT_CONF="/etc/wpa_supplicant.conf"
